src/openms_gui/source/VISUAL/DataSelectionTabs.cpp

```diff
@@ -94,15 +95,24 @@ namespace OpenMS
     if (index != IDENT_IDX) throw Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Tab index is expected to be 1");
     index = addTab(dia_widget_, dia_widget_->objectName());
     if (index != DIAOSW_IDX) throw Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Tab index is expected to be 2");
-    setTabEnabled(SPECTRA_IDX, true);
-    setTabEnabled(IDENT_IDX, false);
-    setTabEnabled(DIAOSW_IDX, false);
+    // make sure initialization was correct
+    assert(tabBar()->count() == tab_ptrs_.size());
 
     // switch between different view tabs
     connect(this, &QTabWidget::currentChanged, this, &DataSelectionTabs::currentTabChanged);
     connect(this, &QTabWidget::tabBarDoubleClicked, this, &DataSelectionTabs::tabBarDoubleClicked);
   }
 
+  LayerData* getCurrentLayerData(TOPPViewBase* tv)
```

Comment:
I think it is bad design to go back to the parent here and request the Layer

Reply:
Not your fault though :)

Reply:
Mhh.. with the way we currently do things, I don't see what we can do about it.
One could rewrite this using signal/slot, but that only makes it less obvious.

Reply:
I think signal/slots would probably be better because it would move the design to a notify/observer model and not to the current "you need to update yourself go pull your data from the god object" model ;)